label_studio/io_storages/base_models.py

```diff
@@ -551,15 +554,16 @@ def save_all_annotations(self):
         self.info_set_in_progress()
         self.cached_user = self.project.organization.created_by
 
-        for annotation in Annotation.objects.filter(project=self.project).iterator(
-            chunk_size=settings.STORAGE_EXPORT_CHUNK_SIZE
-        ):
-            annotation.cached_user = self.cached_user
-            self.save_annotation(annotation)
+        max_workers = min(32, (os.cpu_count() or 4) * 4)
```

Comment:
Do you see a need to let storage instances override this? (for different cloud providers, or in the future, provider-specific handling of max concurrent writes based on storage details) If so, it can be `self.max_workers`